### ISSUE TITLE
how can i save the model as same as your released lstm model?

### ISSUE THREAD

**Issue by Courageux-J** [jiangsutx/SRN-Deblur#21]

**Comment by jiangsutx** [comment]
You can refer to the documents or this article.
https://blog.metaflow.fr/tensorflow-how-to-freeze-a-model-and-serve-it-with-a-python-api-d4f3596b3adc

It is called freeze_graph to convert all variables to constant for inference use only.

**Comment by Courageux-J** [comment]
Could you give me the (lstm,gray and rgb) model before released ,i want to continue training on the basis of your model

**Comment by jiangsutx** [comment]
You can load these released files, and continue training. They just do not have intermediate gradients.

**Comment by Courageux-J** [comment]
yes,i  had a try with your released files,but the error happened

NotFoundError (see above for traceback): Restoring from checkpoint failed. This is most likely due to a Variable name or other graph key that is missing from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:
Key Variable not found in checkpoint
	 [[Node: save/RestoreV2 = RestoreV2[dtypes=[DT_INT32, DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, ..., DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT], _device="/job:localhost/replica:0/task:0/device:CPU:0"](_arg_save/Const_0_0, save/RestoreV2/tensor_names, save/RestoreV2/shape_and_slices)]]

**Comment by jiangsutx** [comment]
You can first do a random initialization, just like training from scratch.
Then you load our released model. And continue tranining

After that, I think you may get rid of these errors.

**Comment by Courageux-J** [comment]
i can load my model successfully,but when load the released model the error happend.i used the following code .it's not ok in the stage of training .

        checkpoint_dir = './checkpoints/lstm'
        ckpt_name = 'deblur.model-523000'
        self.saver.restore(sess, os.path.join(checkpoint_dir, ckpt_name))     # error happened

**Comment by Courageux-J** [comment]
thank you very much,the problem has been solved. the model i saved includes the gradients
